src/components/application_manager/test/commands/mobile/delete_file_test.cc

```diff
@@ -0,0 +1,239 @@
+using am::commands::DeleteFileRequest;
+using am::commands::DeleteFileResponse;
+using am::commands::MessageSharedPtr;
+using am::event_engine::Event;
+using am::MockMessageHelper;
+
+typedef SharedPtr<DeleteFileRequest> DeleteFileRequestPtr;
+typedef SharedPtr<DeleteFileResponse> DeleteFileResponsePtr;
+typedef NiceMock<
+    ::test::components::application_manager_test::MockHMICapabilities>
+    MockHMICapabilities;
+
+ACTION_TEMPLATE(SetArgPointer,
+                HAS_1_TEMPLATE_PARAMS(int, k),
+                AND_1_VALUE_PARAMS(vec)) {
+  *std::tr1::get<k>(args) = *vec;
+}
+
+MATCHER_P(CheckMessageResultCode, result_code, "") {
+  return (*arg)[am::strings::msg_params][am::strings::result_code].asInt() ==
+         result_code;
+}
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kCorrelationId = 10u;
+const int32_t kMenuId = 5;
+}  // namespace
+
+class DeleteFileRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  void SetUp() OVERRIDE {
+    message_ = CreateMessage();
+    command_ = CreateCommand<DeleteFileRequest>(message_);
+    app_ = CreateMockApp();
+  }
+  DeleteFileRequestPtr command_;
+  MessageSharedPtr message_;
+  MockAppPtr app_;
+};
+
+class DeleteFileResponseTest : public CommandsTest<CommandsTestMocks::kIsNice> {
+};
+
+TEST_F(DeleteFileRequestTest, Run_InvalidApp_UNSUCCESS) {
+  MockAppPtr invalid_app;
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(invalid_app));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL));
+  EXPECT_CALL(app_mngr_, get_settings()).Times(0);
+
+  command_->Run();
+}
+
+TEST_F(DeleteFileRequestTest, Run_HMILevelNone_UNSUCCESS) {
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(app_));
+  EXPECT_CALL(*app_, hmi_level())
+      .WillOnce(Return(am::mobile_api::HMILevel::HMI_NONE));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+  const uint32_t kNum = 0;
+  EXPECT_CALL(app_mngr_settings_, delete_file_in_none())
+      .WillOnce(ReturnRef(kNum));
+  EXPECT_CALL(*app_, delete_file_in_none_count()).WillOnce(Return(1));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(CheckMessageResultCode(mobile_apis::Result::REJECTED),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command_->Run();
+}
+
+TEST_F(DeleteFileRequestTest, Run_ValidFileName_SUCCESS) {
+  const std::string kFileName = "test_file.txt";
+  EXPECT_TRUE(file_system::CreateFile(kFileName));
+  (*message_)[am::strings::msg_params][am::strings::sync_file_name] = kFileName;
+
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(app_));
+  EXPECT_CALL(*app_, hmi_level())
+      .WillOnce(Return(am::mobile_api::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+  const std::string kFullFilePath = file_system::CurrentWorkingDirectory();
```

Comment:
@AGritsevich correct the const name, and in order not to spam your PR with the same comment, please just check everywhere